feat(javascript/rest-sveltekit): add SvelteKit example

joshnuss opened the pull request:
This adds an example of using Prisma with SvelteKit.

It is adapted from the `rest-nextjs` example.

joshnuss committed:
Adds SvelteKit example

ruheni commented on javascript/rest-sveltekit/README.md:
````diff
@@ -0,0 +1,300 @@
+
+```bash
+npx prisma migrate dev --name init
+```bash
````
```suggestion
```
```

joshnuss committed:
Fixup

joshnuss committed:
Fixup: keep import similar to nextjs example

joshnuss committed:
Fixup: remove incorrect comment

joshnuss committed:
Fixup: disable engine-strict to allow testing on nodejs 14.x

joshnuss committed:
Fixup: add required attribute

ruheni reviewed (changes requested):
Hi @joshnuss 👋🏽

Thank you for your contribution to the Prisma Examples. 

Here are a few improvements I'd like you to make before we merge this:
- Add the seed script in the `package.json` file to match the instructions in the README
- The anchor tags have the default color. Kindly update the color to match the colors with the other fullstack examples
- Add a test in `.github/tests/javascript/rest-sveltekit` – You can copy and paste the test from `github/tests/javascript/rest-nextjs` for how to do it
- Our READMEs are auto-generated based on the templates in `.github/readmes`. Create a template for the README in `.github/readmes/javascript/rest-sveltekit`. I would recommend looking at the `rest-nuxtjs` example README and adapting it for this example. Once you've made the change, navigate to `./github/tools`, run `npm install`, and finally `npm run build-readmes`
- When I ran `npm run dev` the first time, I ran into the following error:
```

$ npm run dev

> rest-sveltekit@0.0.1 dev
> vite dev

failed to load config from /workspaces/prisma-examples/javascript/rest-sveltekit/vite.config.js
error when starting dev server:
Error: Vite version 2 is no longer supported. Please upgrade to version 3
    at check_vite_version (file:///workspaces/prisma-examples/javascript/rest-sveltekit/node_modules/@sveltejs/kit/dist/vite.js:3177:9)
    at kit (file:///workspaces/prisma-examples/javascript/rest-sveltekit/node_modules/@sveltejs/kit/dist/vite.js:2931:2)
    at sveltekit (file:///workspaces/prisma-examples/javascript/rest-sveltekit/node_modules/@sveltejs/kit/dist/vite.js:2828:23)
    at file:///workspaces/prisma-examples/javascript/rest-sveltekit/vite.config.js?t=1657872771313:5:12
    at ModuleJob.run (node:internal/modules/esm/module_job:197:25)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:337:24)
    at async importModuleDynamicallyWrapper (node:internal/vm/module:437:15)
    at async loadConfigFromFile (/workspaces/prisma-examples/javascript/rest-sveltekit/node_modules/vite/dist/node/chunks/dep-c9998dc6.js:61917:31)
    at async resolveConfig (/workspaces/prisma-examples/javascript/rest-sveltekit/node_modules/vite/dist/node/chunks/dep-c9998dc6.js:61450:28)
```
I was able to fix it by updating the dependencies.
- I couldn't reproduce this error, but I saw this on my terminal a couple of times when I selected a draft _I think_. The app worked, but I think the error could give a false warning
```
Invalid `prisma.post.findUnique()` invocation:

{
  where: {
    id: NaN
        ~~~
  },
  include: {
    author: true
  }
}

Argument id: Got invalid value NaN on prisma.findUniquePost. Provided Float, expected Int.


Error: 
Invalid `prisma.post.findUnique()` invocation:

{
  where: {
    id: NaN
        ~~~
  },
  include: {
    author: true
  }
}

Argument id: Got invalid value NaN on prisma.findUniquePost. Provided Float, expected Int.


    at Document.validate (/workspaces/prisma-examples/javascript/rest-sveltekit/node_modules/@prisma/client/runtime/index.js:48195:20)
    at PrismaClient._executeRequest (/workspaces/prisma-examples/javascript/rest-sveltekit/node_modules/@prisma/client/runtime/index.js:50632:17)
    at consumer (/workspaces/prisma-examples/javascript/rest-sveltekit/node_modules/@prisma/client/runtime/index.js:50569:23)
    at /workspaces/prisma-examples/javascript/rest-sveltekit/node_modules/@prisma/client/runtime/index.js:50573:76
    at runInChildSpan (/workspaces/prisma-examples/javascript/rest-sveltekit/node_modules/@prisma/client/runtime/index.js:49780:12)
    at /workspaces/prisma-examples/javascript/rest-sveltekit/node_modules/@prisma/client/runtime/index.js:50573:20
    at AsyncResource.runInAsyncScope (node:async_hooks:201:9)
    at PrismaClient._request (/workspaces/prisma-examples/javascript/rest-sveltekit/node_modules/@prisma/client/runtime/index.js:50572:86)
    at requestFn (/workspaces/prisma-examples/javascript/rest-sveltekit/node_modules/@prisma/client/runtime/index.js:49316:65)
    at /workspaces/prisma-examples/javascript/rest-sveltekit/node_modules/@prisma/client/runtime/index.js:49329:18
```


Looking forward to merging this. 💚

joshnuss commented:
Thanks for the review @ruheni! I will resolve these points over the next few days.

joshnuss committed:
Adds seed command to package.json

joshnuss committed:
Updates seeds to use es import

joshnuss committed:
Adjust link color

joshnuss committed:
Use absolute path for favicon to avoid messing with /p/[id] endpoint

ruheni commented:
Hey @joshnuss, any updates on the example?

joshnuss commented:
Hey @ruheni,

I fixed a few things:

- Link color was updated.
- Added the seed script in the package.json file to match the instructions in the README
- The error `Invalid prisma.post.findUnique() invocation` is fixed. It was caused by the `favicon.png` using a relative path for some reason. That caused `/p/favicon.png` to be fetched which caused an error. 

A few notes:

- The vite error doesn't happen on my machine, but I think we should be using vite 3. so i'll update it.
- The `src/routes` folder structure is currently [undergoing an overhaul in SvelteKit](https://github.com/sveltejs/kit/discussions/5748), so we might want to wait until that is released before merging this PR

ruheni commented:
Hey @joshnuss 👋🏽

Any updates on the PR?

joshnuss commented:
Hi @ruheni,

I've not had time to complete it.

Feel free to close it, and I'll re-open it when I have some free time.

Best,
Josh

ruheni commented:
Hey @joshnuss 

Sounds good. If you need any help, don't hesitate to give me a ping.

Looking forward to the example when you get the time to continue with the PR 🙂
